src/tlo/analysis/utils.py

```diff
@@ -610,6 +610,7 @@ def filter_treatments(_treatments: Iterable[str], depth: int = 1) -> List[str]:
             [
                 "".join(f"{x}_" for i, x in enumerate(t.split('_')) if i < depth).rstrip('_') + '_*'
                 for t in set(_treatments)
+                if t # In the event an abstract base class is detected, that does not set TREATMENT_ID by default
```

Comment:
As far as I can tell, this portion of the code uses `iterpackage` to read the package's source code and detect (HSI-event specific) classes that are defined in the various disease modules.

It then looks through the HSI-event classes it decides it has found to determine all the possible `treatment_id`s that are provided. However, some classes like `HSI_BaseGenericFirstAppt` do not set a `treatment_id` by default (since it is overwritten by derived classes). In the list (technically `set`) of `treatment_ids` that is being assembled, this causes an empty string to be added to the list. The test then fails since the list of `treatment_id`s now contains this empty string, whilst the reference set of `treatment_id`s does not. Adding this `if t` check ensures that these empty strings are filtered out.

Reply:
Ah that makes sense - thanks for the explanation. I should have expanded the code context more around the changed snippet!